truss/truss_handle/build.py

```diff
@@ -24,36 +28,62 @@
 
 
 def _populate_target_directory(
-    config: TrussConfig,
-    target_directory_path: Optional[str] = None,
-    template: str = "custom",
-    populate_dirs: bool = True,
+    config: TrussConfig, target_directory: str, python_dx: bool = False
 ) -> Path:
-    target_directory_path_typed = None
-    if target_directory_path is None:
-        target_directory_path_typed = build_truss_target_directory(template)
-    else:
-        target_directory_path_typed = Path(target_directory_path)
-        target_directory_path_typed.mkdir(parents=True, exist_ok=True)
+    target_directory_path = Path(target_directory)
+    target_directory_path.mkdir(parents=True, exist_ok=True)
 
-    if populate_dirs:
+    if not python_dx:
         # Create data dir
-        (target_directory_path_typed / config.data_dir).mkdir()
+        (target_directory_path / config.data_dir).mkdir()
 
         # Create bundled packages dir
         # TODO: Drop by default
-        (target_directory_path_typed / config.bundled_packages_dir).mkdir()
+        (target_directory_path / config.bundled_packages_dir).mkdir()
 
         # Create model module dir
-        model_dir = target_directory_path_typed / config.model_module_dir
-        template_path = TEMPLATES_DIR / template
-        copy_tree_path(template_path / "model", model_dir)
+        model_dir = target_directory_path / config.model_module_dir
+        copy_tree_path(TRADITIONAL_CUSTOM_TEMPLATE_DIR / "model", model_dir)
+
+        # Write config
+        with (target_directory_path / CONFIG_FILE).open("w") as config_file:
+            yaml.dump(config.to_dict(verbose=False), config_file)
+    else:
+        copy_tree_path(PYTHON_DX_CUSTOM_TEMPLATE_DIR, target_directory_path)
 
-    # Write config
-    with (target_directory_path_typed / CONFIG_FILE).open("w") as config_file:
-        yaml.dump(config.to_dict(verbose=False), config_file)
+        # Hack: We want to place the user provided model name into generated code. Until
```

Comment:
I really don't love this, but didn't want to overengineer a way to get the user provided model name into the generated file